scripts/lib/email-mirror-postfix

```diff
@@ -99,12 +101,16 @@ def send_email_mirror(
     if not rcpt_to:
         print("5.1.1 Bad destination mailbox address: No missed message email address.")
         exit(posix.EX_NOUSER)
-    msg_text = sys.stdin.read(MAX_ALLOWED_PAYLOAD + 1)
-    if len(msg_text) > MAX_ALLOWED_PAYLOAD:
+    msg_bytes = sys.stdin.buffer.read(MAX_ALLOWED_PAYLOAD + 1)
+    if len(msg_bytes) > MAX_ALLOWED_PAYLOAD:
         # We're not at EOF, reject large mail.
         print("5.3.4 Message too big for system: Max size is 25MiB")
         exit(posix.EX_DATAERR)
 
+    msg_text = email.message_from_bytes(
+        msg_bytes, policy=email.policy.default,
+    ).as_string()
+
```

Comment:
Does this change not require any changes in the code that handles the request?  Maybe some detail on the case this extra logic is before in a comment would be helpful?

Reply:
We’re just parsing the message and re-serializing it to a 7-bit-clean subset of the same format; messages with `Content-Transfer-Encoding: 8bit` get converted to `Content-Transfer-Encoding: base64`.

Though perhaps it would be better to fix the `email_mirror_message` endpoint API to be binary-safe, if we’re willing to change it—the JSON wrapper is pointless.

Reply:
I'm fine with changing it; there's no reason we need to keep that interface particularly stable.

Reply:
Okay, now we just base64 encode the entire message on the outside.